<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MaemoSettingsWidget</class>
 <widget class="QWidget" name="MaemoSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>596</width>
    <height>352</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Maemo Device Configurations</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="configurationLabel">
         <property name="text">
          <string>Configuration:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="configurationComboBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="editable">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QFrame" name="frame">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QWidget" name="detailsWidget" native="true">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="deviceNameLabel">
             <property name="text">
              <string>Name</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="nameLineEdit"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="deviceTypeLabel">
             <property name="text">
              <string>Device type:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QWidget" name="deviceTypeButtonsWidget" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="margin">
               <number>0</number>
              </property>
              <item>
               <widget class="QRadioButton" name="deviceButton">
                <property name="text">
                 <string>Remote device</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="simulatorButton">
                <property name="text">
                 <string>Maemo emulator</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="authTypeLabel">
             <property name="text">
              <string>Authentication type:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QWidget" name="authTypeButtonsWidget" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <property name="margin">
               <number>0</number>
              </property>
              <item>
               <widget class="QRadioButton" name="passwordButton">
                <property name="text">
                 <string>Password</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="keyButton">
                <property name="text">
                 <string>Key</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="hostNameLabel">
             <property name="text">
              <string>Host name:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLineEdit" name="hostLineEdit">
             <property name="placeholderText">
              <string>IP or host name of the device</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="portsLabel">
             <property name="text">
              <string>Ports:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="sshPortLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>SSH:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sshPortSpinBox">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>65535</number>
               </property>
               <property name="value">
                <number>22</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="gdbServerLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Gdb server:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="gdbServerPortSpinBox">
               <property name="maximum">
                <number>65535</number>
               </property>
               <property name="value">
                <number>10000</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="connectionTimeoutLabel">
             <property name="text">
              <string>Connection timeout:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QSpinBox" name="timeoutSpinBox">
               <property name="suffix">
                <string>s</string>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="singleStep">
                <number>10</number>
               </property>
               <property name="value">
                <number>1000</number>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="userNameLabel">
             <property name="text">
              <string>Username:</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLineEdit" name="userLineEdit"/>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="passwordLabel">
             <property name="text">
              <string>Password:</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLineEdit" name="pwdLineEdit">
             <property name="echoMode">
              <enum>QLineEdit::Password</enum>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="keyLabel">
             <property name="text">
              <string>Private key file:</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="Utils::PathChooser" name="keyFileLineEdit" native="true"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="addConfigButton">
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeConfigButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Remove</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="testConfigButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Test</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="generateKeyButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Generate SSH Key ...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="deployKeyButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Deploy Public Key ...</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
   <slots>
    <signal>editingFinished()</signal>
    <signal>browsingFinished()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>deviceButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deviceTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hostLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>hostNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>userLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>userNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>419</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pwdLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>passwordEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>419</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulatorButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deviceTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>addConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>465</x>
     <y>27</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deleteConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>passwordButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>authenticationTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>240</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyFileLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>keyFileEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>419</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyFileLineEdit</sender>
   <signal>browsingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>keyFileEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>419</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>testConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>testConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deployKeyButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deployKey()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>authenticationTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>525</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>configNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>330</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generateKeyButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>showGenerateSshKeyDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>585</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>358</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timeoutSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>timeoutEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>227</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timeoutSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>timeoutEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>224</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gdbServerPortSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>gdbServerPortEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gdbServerPortSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>gdbServerPortEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>395</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>590</x>
     <y>251</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sshPortSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>sshPortEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>211</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sshPortSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>sshPortEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>243</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>configNameEditingFinished()</slot>
  <slot>deviceTypeChanged()</slot>
  <slot>hostNameEditingFinished()</slot>
  <slot>sshPortEditingFinished()</slot>
  <slot>timeoutEditingFinished()</slot>
  <slot>userNameEditingFinished()</slot>
  <slot>passwordEditingFinished()</slot>
  <slot>addConfig()</slot>
  <slot>selectionChanged()</slot>
  <slot>deleteConfig()</slot>
  <slot>authenticationTypeChanged()</slot>
  <slot>keyFileEditingFinished()</slot>
  <slot>testConfig()</slot>
  <slot>deployKey()</slot>
  <slot>gdbServerPortEditingFinished()</slot>
  <slot>currentConfigChanged(int)</slot>
  <slot>showGenerateSshKeyDialog()</slot>
 </slots>
</ui>
